### PR TITLE
add conflict checking and multiple people in interviews

### PR DESCRIPTION
**Pull request by viditjain5598**
Adds:
- Multiple users per interview
- Conflict checking

### CODE REVIEW

File: app/controllers/interviews_controller.rb [hunk 1]
```diff
@@ -43,8 +53,19 @@ def destroy
 
   private
 
+  def no_conflicts(members, new_interview)
```
Comment (jaindheerain) [inline app/controllers/interviews_controller.rb:56]: this should not be in controller, rather in model

File: app/controllers/interviews_controller.rb [hunk 2]
```diff
@@ -43,8 +53,19 @@ def destroy
...
+    members.each do |member|
+      @conflicting_interviews = Interview.select("interviews.id").joins(:user_interviews).where(scheduled_time: new_interview.scheduled_time .. new_interview.end_time)
```
Comment (jaindheerain) [inline app/controllers/interviews_controller.rb:58]: suppose there is an inteview with start and end as 5pm and 7pm 
what if i schedule an interview with start and end as 5:30pm and 6:30 pm

File: src/app.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+"use strict";
+
+import router from "./views/pages/router.js";
+import Error404 from "./views/pages/error404.js";
+
+const renderPage = async () => {
+  let view = router.match();
+  const root = document.getElementById('root');
+  if (view != null) {
```
Comment (jaindheerain) [inline src/app.js:9]: !==

**Comment by jaindheerain** [comment]
Don't send old reminder when for the updated interview missing.
User roles is not suported.

File: app/controllers/interviews_controller.rb (hunk 1 above)
Comment (viditjain5598) [reply app/controllers/interviews_controller.rb:56]: Can you elaborate the reason for this?

File: src/app.js (hunk 3 above)
Comment (viditjain5598) [reply src/app.js:9]: This was a work in progress. Shouldn't have been in this PR. belongs in the SPA PR, i.e. pull/3.

**Comment by viditjain5598** [comment]
> Don't send old reminder when for the updated interview missing.
> User roles is not suported.

I couldn't find a way to delete the earlier scheduled interviews (As discussed in an earlier call).

The user roles would required morphing the schema too much, so, I was decided to forgo that in favour of implementing them in further tasks.

File: app/controllers/interviews_controller.rb (hunk 2 above)
Comment (viditjain5598) [reply app/controllers/interviews_controller.rb:58]: That's a case I missed, will rectify soon.

File: app/controllers/interviews_controller.rb (hunk 1 above)
Comment (viditjain5598) [reply app/controllers/interviews_controller.rb:56]: Resolved in https://github.com/viditjain5598/BetterInterviews/pull/4

File: app/controllers/interviews_controller.rb (hunk 2 above)
Comment (viditjain5598) [reply app/controllers/interviews_controller.rb:58]: Resolved in https://github.com/viditjain5598/BetterInterviews/pull/4

**Pull request merged**